vcu/Src/motorController.c

```diff
@@ -45,302 +25,180 @@ float min(float a, float b)
+    // Note - lockout will not disable if inverter is faulted
 
-float limit(float in, float min, float max)
-{
-    if (in > max) {
-        in = max;
-    } else if (in < min) {
-        in = min;
+    if (isLockoutDisabled()) { // disabled
+        // Don't need to release
+        return HAL_OK;
     }
 
-    return min;
+    return sendDisableMC();
 }
 
-HAL_StatusTypeDef sendThrottleValueToMCs(float throttle, int steeringAngle)
-{
-
-	if(!motors_active)
-	{
-        
-		return HAL_OK;
-	}
-
-    float maxTorqueDemand = min(mcRightSettings.DriveTorqueLimit, mcLeftSettings.DriveTorqueLimit);
-
-	float throttleRight = throttle - (throttle * steeringAngle * TORQUE_VECTOR_FACTOR);
-	float throttleLeft = throttle + (throttle * steeringAngle * TORQUE_VECTOR_FACTOR);
-
-    float torqueDemandR = map_range_float(throttleRight, 0, 100, 0, maxTorqueDemand);
-    float torqueDemandL = map_range_float(throttleLeft, 0, 100, 0, maxTorqueDemand);
-
-    TorqueLimitDriveRight = mcRightSettings.DriveTorqueLimit;
-    TorqueLimitBrakingRight = mcRightSettings.BrakingTorqueLimit;
-    TorqueDemandRight = torqueDemandR;
-
-    TorqueLimitDriveLeft = mcLeftSettings.DriveTorqueLimit;
-    TorqueLimitBrakingLeft = mcLeftSettings.BrakingTorqueLimit;
-    TorqueDemandLeft = torqueDemandL;
-
-    SpeedLimitForwardRight = mcRightSettings.ForwardSpeedLimit;
-    SpeedLimitReverseRight = mcRightSettings.ReverseSpeedLimit;
-    InverterCommandRight = mcRightSettings.InverterCommand; // Keep sending enable bridge command
-
-    SpeedLimitForwardLeft = mcLeftSettings.ForwardSpeedLimit;
-    SpeedLimitReverseLeft = mcLeftSettings.ReverseSpeedLimit;
-    InverterCommandLeft = mcLeftSettings.InverterCommand; // Keep sending enable bridge command
-
-    CurrentLimitDschrgInverterRight = mcRightSettings.DischargeCurrentLimit;
-    CurrentLimitChargeInverterRight = mcRightSettings.ChargeCurrentLimit;;
 
-    CurrentLimitDschrgInverterLeft = mcLeftSettings.DischargeCurrentLimit;
-    CurrentLimitChargeInverterLeft = mcLeftSettings.ChargeCurrentLimit;;
+HAL_StatusTypeDef requestTorqueFromMC(float throttle, int steeringAngle) {
 
-    VoltageLimitHighInverterRight = mcRightSettings.HighVoltageLimit;
-    VoltageLimitLowInverterRight = mcRightSettings.LowVoltageLimit;;
+    // Per Cascadia Motion docs, torque requests are sent in Nm * 10
+    float maxTorqueDemand = min(mcSettings.MaxTorqueDemand, mcSettings.DriveTorqueLimit);
+    float scaledTorque = map_range_float(throttle, 0, 100, 0, maxTorqueDemand);
+    uint16_t requestedTorque = scaledTorque * 10;
 
-    VoltageLimitHighInverterLeft = mcLeftSettings.HighVoltageLimit;
-    VoltageLimitLowInverterLeft = mcLeftSettings.LowVoltageLimit;;
+    VCU_INV_Torque_Command = requestedTorque;
+    VCU_INV_Torque_Command = 100; // hardcoded for testing
+    VCU_INV_Speed_Command = TORQUE_MODE_SPEED_REQUEST;
+    VCU_INV_Direction_Command = INVERTER_DIRECTION_FORWARD;
```

Comment:
How easy is it for us to implement reverse? Might have to add an easter egg where we can go backwards in case we are in a sticky situation at comp and to pull a smooth one :DDDD